### PR TITLE
Replacing checkout with switch

### LINKED ISSUE #208: Replace use of git checkout command with git switch or git restore
`git checkout` is a generally confusing command because it has so many use cases. Some sources I have read recommend replacing it with either `git switch` or `git restore` depending on the use case. See https://stackoverflow.com/a/57266005

I  suggest that we replace all occurrences of `git checkout` with the appropriate version of `git switch` or `git restore`. It is worth noting that `git` itself no longer recommends using `git checkout` if you want to discard changes in the working tree, and instead suggests `git restore`. 

I am happy to raise a PR is we get consensus on this.

### PR DESCRIPTION
**Pull request by anenadic**
Fixes #208.

### CODE REVIEW

**Comment by anenadic** [comment]
Any chance you can have a look at this PR @bielsnohr?

**Review by douglowe — APPROVED** [review]
These changes look sensible to me.

File: _episodes/14-collaboration-using-git.md [hunk 1]
```diff
@@ -105,9 +105,9 @@ sequenceDiagram
     Staging Area->>+Local Repository: git commit
     Local Repository->>+Remote Repository: git push
     Remote Repository->>+Local Repository: git fetch
-    Local Repository->>+Staging Area:git checkout
+    Local Repository->>+Staging Area:git checkout / git switch
     Local Repository->>+Staging Area:git merge
-    Remote Repository->>+Working Directory: git pull (shortcut for git fetch followed by git checkout/merge)
+    Remote Repository->>+Working Directory: git pull (shortcut for git fetch followed by git checkout / switch / merge)
```
[the remark this answers is not in the document]
Comment (bielsnohr) [reply _episodes/14-collaboration-using-git.md:110]: I will push a separate commit updating the SVG image below.

**Review by bielsnohr — CHANGES_REQUESTED** [review]
Just a few minor corrections before we merge in.

File: _episodes/14-collaboration-using-git.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (anenadic) [reply _episodes/14-collaboration-using-git.md]: Thank you @bielsnohr

**Comment by anenadic** [comment]
Thank you for catching those few "checkouts" @bielsnohr - do you think it's ready to be merged?

**Review by bielsnohr — APPROVED** [review]
LGTM!

**Pull request merged**